esmvalcore/_recipe.py

```diff
@@ -387,33 +386,54 @@ def _add_fxvar_keys(fx_var_dict, variable):
 
 def _get_correct_fx_file(variable, fx_varname, config_user):
     """Get fx files (searching all possible mips)."""
-    # TODO: allow user to specify certain mip if desired
+    # first see if we have strings or dicts for fx_varname
+    user_defined = False
+    if isinstance(fx_varname, dict):
+        user_defined = True
+        user_fx_mip = fx_varname.get('mip')
+        user_fx_experiment = fx_varname.get('exp')
+        fx_varname = fx_varname.get('short_name')
+
+    # assemble info from master variable
     var = dict(variable)
     var_project = variable['project']
     cmor_table = CMOR_TABLES[var_project]
+    valid_fx_vars = []
 
     # Get all fx-related mips ('fx' always first, original mip last)
     fx_mips = ['fx']
     fx_mips.extend(
         [key for key in cmor_table.tables if 'fx' in key and key != 'fx'])
     fx_mips.append(variable['mip'])
 
+    # force only the mip declared by user
+    if user_defined and user_fx_mip:
+        fx_mips = [
+            user_fx_mip,
+        ]
+
     # Search all mips for available variables
+    # priority goes to user specified mip if available
     searched_mips = []
     for fx_mip in fx_mips:
         fx_variable = cmor_table.get_variable(fx_mip, fx_varname)
         if fx_variable is not None:
             searched_mips.append(fx_mip)
-            fx_var = _add_fxvar_keys(
-                {'short_name': fx_varname, 'mip': fx_mip}, var)
-            logger.debug("For CMIP6 fx variable '%s', found table '%s'",
-                         fx_varname, fx_mip)
+            fx_var = _add_fxvar_keys({
+                'short_name': fx_varname,
+                'mip': fx_mip
+            }, var)
+            if user_defined and user_fx_experiment:
+                fx_var['exp'] = user_fx_experiment
+            logger.debug("For fx variable '%s', found table '%s'", fx_varname,
+                         fx_mip)
             fx_files = _get_input_files(fx_var, config_user)
 
             # If files found, return them
             if fx_files:
-                logger.debug("Found CMIP6 fx variables '%s':\n%s",
-                             fx_varname, pformat(fx_files))
+                valid_fx_vars.append(fx_var)
+                logger.debug("Found fx variables '%s':\n%s", fx_varname,
+                             pformat(fx_files))
                 break
     else:
```

Comment:
Is this `else` statementlinked to a specific `if`? Looks like its preceded by the `for` loop?

Reply:
it is - the `break` case just on top of it - by @schlunma but I actually changed the declaration of `fx_files` so that it's a bit more straightforward

Reply:
It's syntactic sugar of python, see e.g. https://book.pythontips.com/en/latest/for_-_else.html

Reply:
indeed, I killed the sugar tho :grin: - should be ok no?